openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIResponseMapper.java

```diff
@@ -318,54 +326,37 @@ private void setCacheConfigField(CacheConfigAnnotation cacheConfig, String field
     }
 
     /**
-     * This methods for handle the return type has non annotations.
+     * This method for handle the return type has non annotations.
      *
      * @return {@link io.swagger.v3.oas.models.responses.ApiResponses} for operation.
      */
-    private ApiResponses getAPIResponses(OperationAdaptor operationAdaptor, ApiResponses apiResponses,
-                                         Node typeNode, Optional<String> customMediaPrefix, Map<String, Header> headers)
-            throws OpenApiConverterException {
+    private Optional<ApiResponses> getAPIResponses(OperationAdaptor operationAdaptor, ApiResponses apiResponses,
```

Comment:
Do we need to keep a separate class `ApiResponses`? Isn't it possible to use a list/array for the same purpose?

Reply:
`ApiResponses` is swagger libraries defined data model 🙂